IOPool/TFileAdaptor/src/TFileAdaptor.cc

```diff
@@ -68,32 +71,26 @@ bool TFileAdaptor::native(char const* proto) const {
 }
 
 TFileAdaptor::TFileAdaptor(edm::ParameterSet const& pset, edm::ActivityRegistry& ar)
-    : enabled_(true),
-      doStats_(true),
+    : enabled_(pset.getUntrackedParameter<bool>("enable")),
+      doStats_(pset.getUntrackedParameter<bool>("stats")),
       enablePrefetching_(false),
-      cacheHint_("auto-detect"),
-      readHint_("auto-detect"),
-      tempDir_(),
-      minFree_(0),
+      // values set in the site local config or in SiteLocalConfigService override
+      // any values set here for this service.
+      // These parameters here are needed only for backward compatibility
+      // for WMDM tools until we switch to only using the site local config for this info.
+      cacheHint_(pset.getUntrackedParameter<std::string>("cacheHint")),
+      readHint_(pset.getUntrackedParameter<std::string>("readHint")),
+      tempDir_(pset.getUntrackedParameter<std::string>("tempDir")),
```

Comment:
So in theory, the old `f->tempPath()` and the new `StorageFactory::defaultTempDir()` would not have to agree as something could have called `setTempDir()` earlier. But it appears in practice that this code is the only one that calls `setTempDir()` (?).